lib/sections/driver.js

```diff
@@ -1,60 +1,89 @@
-                  if(started === 0 && typeof callback === 'function'){ callback(null, matches); }
-                });
-              }
-            }
+      if(args.bus_class_proxy === '::QueueBus::Driver'){   
+        driverPerform(self, args, callback); 
+      }else if(args.bus_class_proxy === '::QueueBus::Rider'){    
+        callback(new Error('TODO: ::QueueBus::Rider')); // TODO!
+      }else if(args.bus_class_proxy === '::QueueBus::Publisher'){
+        publisherPerform(self, args, callback); 
+      }else if(args.bus_class_proxy === '::QueueBus::Heartbeat'){
+        callback(new Error('TODO: ::QueueBus::Heartbeat')); // TODO!
+      }else{
+        callback(new Error('Missing bus_class_proxy or unknown method (' + args.bus_class_proxy + ')'));
+      }
+    } 
+  };
+};
+
+var publisherPerform = function(self, args, callback){
+  self.publish(args.bus_event_type, args, function(err){
+    callback(err, self.options.busClassKey);
+  });
+};
+
+var driverPerform = function(self, args, callback){
+  var started = 0;
+  var matches = [];
+
+  var end = function(err){
+    if(started === 0 && typeof callback === 'function'){ 
+      callback(err, matches);
+    }
+  };
+
+  self.subscriptions(function(err, subscriptions, count){
+    if(err){
+      end(err);
+    }else if(count === 0){
+      end();
+    }else{    
+      for(var app in subscriptions){
+        for(var i in subscriptions[app]){
+          started++;
+          var subscription = subscriptions[app][i];
+          var matched = subscriptionMatch(args, subscription);
+          if(matched === true){
+            matches.push(subscription.queue_name);
+            var payload = driverMetadata(args, subscription.queue_name, app, i, subscription.class, args.bus_event_type);
+            self.queueObject.enqueue(subscription.queue_name, subscription.class, payload, function(err, toRun){
+              started--;
+              end(err);
+            });
+          }else{
+            process.nextTick(function(){
+              started--;
+              end();
+            });
           }
         }
-      });
-    } 
-  }
-}
+      }
+    }
+  });
+};
 
 var driverMetadata = function(args, bus_rider_queue, bus_rider_app_key, bus_rider_sub_key, bus_rider_class_name, bus_event_type){
-  var payload = {}
+  var payload = {};
 
-  payload['bus_driven_at']        = utils.timestamp();
-  payload['bus_rider_queue']      = bus_rider_queue;
-  payload['bus_rider_app_key']    = bus_rider_app_key;
-  payload['bus_rider_sub_key']    = bus_rider_sub_key;
-  payload['bus_rider_class_name'] = bus_rider_class_name;
-  payload['bus_event_type']       = bus_event_type;
+  payload.bus_driven_at        = utils.timestamp();
+  payload.bus_rider_queue      = bus_rider_queue;
+  payload.bus_rider_app_key    = bus_rider_app_key;
+  payload.bus_rider_sub_key    = bus_rider_sub_key;
+  payload.bus_rider_class_name = bus_rider_class_name;
+  payload.bus_event_type       = bus_event_type;
```

Comment:
@bleonard from a cursory glance, it seems like the metadata has not changed... am I missing anything?